scipy/special/_basic.py

```diff
@@ -2426,7 +2426,10 @@ def factorial2(n, exact=False):
     """
     if exact:
         if n < -1:
-            return 0
+            if n % 2 != 0:
+                return (factorial2(n + 2, exact)) / (n + 2)
+            else:
+                return ValueError
```

Comment:
We will want to add an informative message. And this would need to be tested with `pytest.raises(ValueError, match=msg)`